geoLocationCloud.js

```diff
@@ -56,6 +56,11 @@ class GeoLocationCloud extends Engine {
 
       cloudData = JSON.parse(cloudData);
 
+      if (cloudData.device && cloudData.device.geolocation === null && cloudData.device.geolocationnullreason) {
+        console.warn("WARNING: geolocation not populated. " + cloudData.device.geolocationnullreason + "\n" +
```

Comment:
Only change I'd suggest here is the use of the log rather than console to give the user more options.
So `flowData.pipeline.log('warn', ...)` rather than `console.warn(...)`.
I'll be including this change in the next release, so I'll make the change internally while merging.